Library/Services/WorkflowManager.php

```diff
@@ -26,47 +27,262 @@ public function __construct(string $workflowConfigFilePath)
     }
 
-    public function init(string $workflowProcessId) {
+    /**
+     * Initialize workflow
+     *
+     * @param string $workflowProcessId
+     */
+    public function init(string $workflowProcessId)
+    {
         $this->getDI()->get('logger')->debug(json_encode(func_get_args()));
-        $this->tmpStorage[$workflowProcessId] = [ 
-            'workflow' => $this->config['step'],
+        $this->tmpStorage[$workflowProcessId] = [
+            'workflow' => ['steps' => $this->config['steps']->toArray()],
             'status' => self::STATUS_STARTED,
             'currentStepPos' => -1,
             'initializedAt' => date('Y-m-d H:i:s'),
             'updatedAt' => date('Y-m-d H:i:s')
         ];
     }
 
-    public function getStatus(string $workflowProcessId) : string {
+    /**
+     * Set workflow status
+     *
+     * @param string $workflowProcessId
+     * @param string $status
+     */
+    public function setStatus(string $workflowProcessId, string $status)
+    {
+        $this->getDI()->get('logger')->debug(json_encode(func_get_args()));
+        if (isset($this->tmpStorage[$workflowProcessId])) {
+            $this->tmpStorage[$workflowProcessId]['status'] = $status;
+        }
+    }
+
+    /**
+     * Get workflow status
+     *
+     * @param string $workflowProcessId
+     * @return string
+     */
+    public function getStatus(string $workflowProcessId) : string
+    {
         $this->getDI()->get('logger')->debug(json_encode(func_get_args()));
         if (!isset($this->tmpStorage[$workflowProcessId]) || empty($this->tmpStorage[$workflowProcessId]['status'])) {
             return self::STATUS_NO_STARTED;
         }
         return $this->tmpStorage[$workflowProcessId]['status'];
     }
 
-    public function getNextStepTaskList(string $workflowProcessId) : array {
-        // Check WF status
+    /**
+     * Go to next step
+     *
+     * @param string $workflowProcessId
+     */
+    public function initNextStep(string $workflowProcessId)
+    {
+        echo PHP_EOL . '>' . __METHOD__ . ' : ' . json_encode(func_get_args());
+        $this->tmpStorage[$workflowProcessId]['currentStepPos']++;
+    }
+
+    /**
+     * Get next step if it exists
+     *
+     * @param string $workflowProcessId
+     * @return array
+     */
+    public function getNextStepList(string $workflowProcessId) : array
+    {
         $this->getDI()->get('logger')->debug(json_encode(func_get_args()));
         $nextStepPos = $this->tmpStorage[$workflowProcessId]['currentStepPos'] + 1;
-        $this->getDI()->get('logger')->debug($nextStepPos);
-        $stepHash = $this->config->steps[$nextStepPos]->toArray();
-        // Deals w/ parallelized task xxx to unitest
-        if (array_keys($stepHash) !== array_keys(array_keys($stepHash))) {
-            return [$stepHash];
+
+        // Manage case when there is no more step to run
+        if(empty($this->config->steps[$nextStepPos])) {
```

Comment:
this has already been moved in a dedicated func..
This PR sounds not up to date